src/layouts/index.js

```diff
@@ -1,19 +1,24 @@
-import React from "react";
-
+import React, { useEffect } from "react";
+import { useTranslation } from "react-i18next"
 import "../global.css";
 
 import Navigation from "../components/navigation";
-import Welcome from "../components/welcome";
 
 import withI18next from "../i18n/withI18next";
 
-const Layout = ({ children }) => {
+const Layout = ({ children, location, pageContext: { locale }}) => {
+
+  const { i18n } = useTranslation()
+
+  useEffect(() => {
+    i18n.changeLanguage(locale)
+  }, [location, i18n, locale])
```

Comment:
What's the idea in mind of adding `i18n` in the dependencies list for the side effect?

Reply:
Ah, the build of gatsby we are using has eslint which gives this warning:
```
warn ESLintError:
/Users/dc/Projects/finbb-frontpage/src/layouts/index.js
  16:6  warning  React Hook useEffect has missing dependencies: 'i18n' and 'locale'. Either include them or remove the dependency array  react-hooks/exhaustive-deps
```

After a [bit of research](https://github.com/facebook/create-react-app/issues/6903) the recommendation is that you should include all dependencies that are inside the useEffect hook inside that second array, as they can change from the outside too, which in theory should change how they are used inside the hook.

Reply:
Alright, thanks, that's cool :)
I personally don't like putting whole objects in a dep list, but if the linter say so :P

Reply:
Sure, but think of it this way. It's used inside the hook, so it is a dependency.